htcrystalball/examine.py

```diff
@@ -161,6 +160,15 @@ def check_slots(static: list, dynamic: list, gpu: list, n_cpus: int,
 
 
 def default_preview(slot_name: str, slot_type: str) -> dict:
+    """
+    Defines the default dictionary for slots that don't fit the job
+    Args:
+        slot_name:
+        slot_type:
+
+    Returns:
```

Comment:
Either write a description for the args and return value, or leave them out. I'd suggest leaving them out, as the argument names and code are pretty self-explanatory. If you want to fill them up, it'd look like this:

```
"""
Defines the default dictionary for slots that don't fit the job.

Args:
    slot_name (str): the name of the cpu slot, e.g. "cpu1"
    slot_type (str): the type of cpu slot, allowed {'dynamic', 'static'}

Returns: 
    dict: default values for a previewed slot

"""
```

Reply:
This is my primary criticism of the PR, is these boilerplate (but ultimately empty) doc strings. They should either be usefully populated, or removed.